File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        match nums.is_empty() {
            true => return 0,
            false => {
                let mut prev = 0;
                for i in 1..nums.len() {
                    if nums[prev] != nums[i] {
                        prev = prev + 1;
                        nums[prev] = nums[i];
                    }
                }
                return (prev + 1) as i32
            }
        }
    }
}

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 0;
        int n = nums.size();
        
        for(int j = 1;j < n; j++) {
            if(nums[i] != nums[j]) {
                i++;
                nums[i] = nums[j];
            }
        }
        
        nums.erase(nums.begin()+i+1, nums.end());
        return i+1;
    }
};